giraffe/src/components/AnnotationHoverLayer.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {FunctionComponent} from 'react'
+import {AnnotationMark} from '../types'
+import {AnnotationLayerProps} from './AnnotationLayer'
+import {AnnotationTooltip} from './AnnotationTooltip'
+
+interface AnnotationHoverLayerProps extends AnnotationLayerProps {
+  annotationPositions: AnnotationMark[]
+  boundingReference: DOMRect
+  hoverRowIndices: number[]
+  width: number
+  height: number
+}
```

Comment:
No, this has nothing to do with it being an svg. Even as a canvas, the hovering would require AnnotationHoverLayer to be aware of the width and height of the plot. The reason is the placement of the tooltip. Since they are anchored at the top of the graph for 'x' annotations, the height is needed. For 'y' annotations, the tooltip is anchored at the right of the graph, hence width is needed.

Reply:
And technically, we actually didn't end up using the height -- yet. Because the top of the graph is 0 in the coordinate system. I think I will remove the height from the interface until it is actually needed.

The width is still needed.

Reply:
>Even as a canvas, the hovering would require AnnotationHoverLayer to be aware of the width and height of the plot. 

that's kind of the point I was making - since it's an SVG it needs to be aware of both the canvas and have a `DOMRect` bounding reference. The DOMRect is the smell that it's mixing responsibilities. it has to be aware of the lower level canvas construct -and- the higher level DOM construct that canvas shouldn't really need to know about. in other words, canvas rendered lines aren't concerned with its contained, but the svg annotation is...